spotify-artist-viewer/Models/SAArtistTableViewController.m

```diff
@@ -11,100 +11,60 @@
 
-
-
 @interface SAArtistTableViewController ()
-//@property (weak, nonatomic) IBOutlet UISearchBar *searchBar;
-
-
+@property (strong, nonatomic) SARequestManager *manager;
 @end
 
 @implementation SAArtistTableViewController
-//connecting this view controller with the Artist object (importing SAArtist, too)
-@synthesize artistArray;
-@synthesize searchResults;
-@synthesize rowNum;
-
 
 - (void)viewDidLoad {
     [super viewDidLoad];
+    self.artists = [NSMutableArray new];
     [self.searchBar setDelegate:self];
-    [self.tableView reloadData];
-   
-
+    self.manager = [SARequestManager sharedManager];
 }
 
 - (void)didReceiveMemoryWarning {
     [super didReceiveMemoryWarning];
-    // Dispose of any resources that can be recreated.
 }
 
 #pragma mark - Text Change
 
 -(void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText{
-    SARequestManager *manager = [SARequestManager sharedManager];
-    
 
-    [manager getArtistWithQuery:searchText success:^(NSArray *artists)  {
-       [self.tableView reloadData];
-        self.artistArray = artists;
+    [self.manager getArtistWithQuery:searchText success:^(NSArray *artists)  {
+      
+        self.artists = [artists mutableCopy];
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [self.tableView reloadData];
+        });
         
      } failure:^(NSError *error) {
          NSLog(@"block failure");
     }];
     
-    
-    
 }
 
 #pragma mark - Button cell pressing
 
-//- (IBAction)buttonPressed:(id)sender
-//{
-//
-//    [self performSegueWithIdentifier:@"NextScreen" sender:sender];
-//}
-
-//-(IBAction)did
-
 -(void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
-    NSInteger row = [indexPath row];
-    rowNum = row;
-    NSLog(@"RowNum - %d", rowNum);
-    [self performSegueWithIdentifier:@"NextScreen" sender:nil];
- 
-    
+    [self performSegueWithIdentifier:@"NextScreen" sender:indexPath];
 }
 
 -(void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender{
     if([[segue identifier ] isEqualToString:@"NextScreen"]){
-        UITableViewCell *selectedCell = (UITableViewCell *)sender;
-        
         SAArtistViewController *svc = [segue destinationViewController];
         SAArtist *artistToBePassed = [[SAArtist alloc]init];
-        
-        
-        artistToBePassed = [self.artistArray objectAtIndex:rowNum];
-        NSLog(@"ArtistName - %@", artistToBePassed.artistName);
-
+        artistToBePassed = [self.artists objectAtIndex:self.tableView.indexPathForSelectedRow.row];
         [svc setVcArtist:artistToBePassed];
-        
-        
     }
-        //get SAArtist
-        // SAArtist *artistToPass;
-        // [stvc
     
 }
 
 #pragma mark - Table view data source
 
-
-
 - (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
-
-    // Return the number of rows in the section.
-    return [self.artistArray count];
+    return [self.artists count];
```

Comment:
When possible use dot notation to access a property. In this case self.artists.count.